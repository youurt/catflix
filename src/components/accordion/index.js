import React, { useState, useContext, createContext } from 'react';
import {
  Container,
  Inner,
  Title,
  Item,
  Header,
  Body,
  Frame,
} from './styles/accordion';

const ToggleContext = createContext();

const Accordion = ({ children, ...restProps }) => {
  return (
    <Container {...restProps}>
      <Inner>{children}</Inner>
    </Container>
  );
};
Accordion.Frame = function AccordionFrame({ children, ...restProps }) {
  return <Frame {...restProps}>{children}</Frame>;
};

Accordion.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Accordion.Item = ({ children, ...restProps }) => {
  const [toggleShow, setToggleShow] = useState(false);
  return (
    <ToggleContext.Provider value={{ toggleShow, setToggleShow }}>
      <Item {...restProps}>{children}</Item>
    </ToggleContext.Provider>
  );
};

Accordion.Header = ({ children, ...restProps }) => {
  const { toggleShow, setToggleShow } = useContext(ToggleContext);

  return (
    <Header
      onClick={() => setToggleShow((prevToggleShow) => !prevToggleShow)}
      {...restProps}
    >
      {children}
      {toggleShow ? (
        <img src="/images/icons/close-slim.png" alt="Close" />
      ) : (
        <img src="/images/icons/add.png" alt="Open" />
      )}
    </Header>
  );
};

Accordion.Body = ({ children, ...restProps }) => {
  const { toggleShow } = useContext(ToggleContext);

  return (
    <Body className={toggleShow ? 'open' : 'closed'} {...restProps}>
      <span>{children}</span>
    </Body>
  );
};

export default Accordion;
