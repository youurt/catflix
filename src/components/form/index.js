import React from 'react';
import {
  Container,
  Base,
  Error,
  Title,
  Text,
  TextSmall,
  Link,
  Input,
  Submit,
} from './styles/form';

const Form = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Form.Base = ({ children, ...restProps }) => {
  return <Base {...restProps}>{children}</Base>;
};

Form.Error = ({ children, ...restProps }) => {
  return <Error {...restProps}>{children}</Error>;
};

Form.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Form.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

Form.TextSmall = ({ children, ...restProps }) => {
  return <TextSmall {...restProps}>{children}</TextSmall>;
};

Form.Link = ({ children, ...restProps }) => {
  return <Link {...restProps}>{children}</Link>;
};

Form.Input = ({ children, ...restProps }) => {
  return <Input {...restProps}>{children}</Input>;
};

Form.Submit = ({ children, ...restProps }) => {
  return <Submit {...restProps}>{children}</Submit>;
};

export default Form;
