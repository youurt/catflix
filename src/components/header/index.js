import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import { Background, Container, Logo, ButtonLink } from './styles/header';

const Header = ({ bg = true, children, ...restProps }) => {
  return bg ? <Background {...restProps}>{children}</Background> : children;
};

Header.Frame = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = ({ children, ...restProps }) => {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = ({ to, ...restProps }) => {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};

export default Header;
