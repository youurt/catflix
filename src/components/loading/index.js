import React from 'react';
import { Spinner, LockBody, Picture, ReleaseBody } from './styles/loading';

const Loading = ({ src, ...restProps }) => {
  return (
    <Spinner {...restProps}>
      <LockBody />
      <Picture src={`/images/users/${src}.png`} data-testid="loading-picture" />
    </Spinner>
  );
};

Loading.ReleaseBody = () => {
  return <ReleaseBody />;
};

export default Loading;
