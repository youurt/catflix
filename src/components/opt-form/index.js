import React from 'react';
import { Container, Input, Button, Text, Break } from './styles/opt-form';

const OptForm = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

OptForm.Input = ({ ...restProps }) => {
  return <Input {...restProps} />;
};

OptForm.Button = ({ children, ...restProps }) => {
  return (
    <Button {...restProps}>
      {children}
      <img src="/images/icons/chevron-right.png" alt="Try Now!" />
    </Button>
  );
};

OptForm.Text = ({ children, ...restProps }) => {
  return <Text {...restProps}>{children}</Text>;
};

OptForm.Break = function OptFormBreak({ ...restProps }) {
  return <Break {...restProps} />;
};

export default OptForm;
