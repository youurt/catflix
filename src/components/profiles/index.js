import React from 'react';
import { Container, Title, List, Picture, Name, Item } from './styles/profiles';

const Profiles = ({ children, ...restProps }) => {
  return <Container {...restProps}>{children}</Container>;
};

Profiles.Title = ({ children, ...restProps }) => {
  return <Title {...restProps}>{children}</Title>;
};

Profiles.List = ({ children, ...restProps }) => {
  return <List {...restProps}>{children}</List>;
};

Profiles.User = ({ children, ...restProps }) => {
  return <Item {...restProps}>{children}</Item>;
};

Profiles.Name = ({ children, ...restProps }) => {
  return <Name {...restProps}>{children}</Name>;
};

Profiles.Picture = ({ src, ...restProps }) => {
  return (
    <Picture
      {...restProps}
      src={src ? `/images/users/${src}.png` : `/images/misc/loading.gif`}
    />
  );
};

export default Profiles;
