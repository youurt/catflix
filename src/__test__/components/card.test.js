import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Card, Player } from '../../components';

const category = 'series';
const slideRows = [
  {
    title: 'Comedy',
    data: [
      {
        genre: 'family',
        maturity: 'PG',
        slug: 'the-jeffersons',
        description:
          'A nouveau riche, African-American family who move into a luxury apartment building develop close, if occasionally fractious, relationships with other tenants.',
        id: '7d2dc83c-c84e-4c27-8d74-5bf6c5469a98',
        title: 'The Jeffersons',
        docId: '2MRuYGPUAtABzj9DZNyU',
      },
    ],
  },
  {
    title: 'Drama',
    data: [
      {
        title: 'Dallas',
        genre: 'drama',
        description:
          'J.R. Ewing, a Texas oil baron, uses manipulation and blackmail to achieve his ambitions, both business and personal. He often comes into conflict with his brother Bobby, his arch-enemy Cliff Barnes and his long-suffering wife Sue Ellen.',
        maturity: '12',
        id: 'cc6696ff-6f35-4216-83fb-d750367c167a',
        slug: 'dallas',
        docId: '3J7BvroMoNOapDOFDePX',
      },
    ],
  },
];

describe('<Card />', () => {
  it('should render the <Card/> with populated data', () => {
    const { container, getByText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item key={item.docId} item={item}>
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(getByText('Comedy')).toBeTruthy();
    expect(getByText('The Jeffersons')).toBeTruthy();
    expect(
      getByText(
        'A nouveau riche, African-American family who move into a luxury apartment building develop close, if occasionally fractious, relationships with other tenants.'
      )
    ).toBeTruthy();

    expect(getByText('Drama')).toBeTruthy();
    expect(getByText('Dallas')).toBeTruthy();
    expect(
      getByText(
        'J.R. Ewing, a Texas oil baron, uses manipulation and blackmail to achieve his ambitions, both business and personal. He often comes into conflict with his brother Bobby, his arch-enemy Cliff Barnes and his long-suffering wife Sue Ellen.'
      )
    ).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render the <Card/> and toggles the card feature', () => {
    const { container, queryByText, getByTestId, getByAltText } = render(
      <Card.Group>
        {slideRows.map((slideItem) => (
          <Card key={`${category}-${slideItem.title.toLowerCase()}`}>
            <Card.Title>{slideItem.title}</Card.Title>
            <Card.Entities>
              {slideItem.data.map((item) => (
                <Card.Item
                  key={item.docId}
                  item={item}
                  data-testid={`${item.slug}-item-feature`}
                >
                  <Card.Image
                    src={`/images/${category}/${item.genre}/${item.slug}/small.jpg`}
                  />
                  <Card.Meta>
                    <Card.SubTitle>{item.title}</Card.SubTitle>
                    <Card.Text>{item.description}</Card.Text>
                  </Card.Meta>
                </Card.Item>
              ))}
            </Card.Entities>
            <Card.Feature category={category}>
              <Player>
                <Player.Button />
                <Player.Video src="/videos/bunny.mp4" />
              </Player>
            </Card.Feature>
          </Card>
        ))}
      </Card.Group>
    );

    expect(queryByText('PG')).toBeFalsy();
    fireEvent.click(getByTestId('the-jeffersons-item-feature'));
    expect(queryByText('PG')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('PG')).toBeFalsy();

    expect(queryByText('12')).toBeFalsy();
    fireEvent.click(getByTestId('dallas-item-feature'));
    expect(queryByText('12')).toBeTruthy();

    fireEvent.click(getByAltText('Close'));
    expect(queryByText('12')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
