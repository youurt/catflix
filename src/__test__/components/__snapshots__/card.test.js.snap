// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render the <Card/> and toggles the card feature 1`] = `
<div
  class="card__Group-sc-1fzidng-2 cuQbQw"
>
  <div
    class="card__Container-sc-1fzidng-1 gcFCET"
  >
    <p
      class="card__Title-sc-1fzidng-0 ighwgO"
    >
      Comedy
    </p>
    <div
      class="card__Entities-sc-1fzidng-12 fjzXQn"
    >
      <div
        class="card__Item-sc-1fzidng-13 hkgFNq"
        data-testid="the-jeffersons-item-feature"
      >
        <img
          class="card__Image-sc-1fzidng-14 bjYtpL"
          src="/images/series/family/the-jeffersons/small.jpg"
        />
        <div
          class="card__Meta-sc-1fzidng-11 gJNAGL"
        >
          <p
            class="card__SubTitle-sc-1fzidng-3 cDBfHO"
          >
            The Jeffersons
          </p>
          <p
            class="card__Text-sc-1fzidng-4 bcshOz"
          >
            A nouveau riche, African-American family who move into a luxury apartment building develop close, if occasionally fractious, relationships with other tenants.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-1fzidng-1 gcFCET"
  >
    <p
      class="card__Title-sc-1fzidng-0 ighwgO"
    >
      Drama
    </p>
    <div
      class="card__Entities-sc-1fzidng-12 fjzXQn"
    >
      <div
        class="card__Item-sc-1fzidng-13 hkgFNq"
        data-testid="dallas-item-feature"
      >
        <img
          class="card__Image-sc-1fzidng-14 bjYtpL"
          src="/images/series/drama/dallas/small.jpg"
        />
        <div
          class="card__Meta-sc-1fzidng-11 gJNAGL"
        >
          <p
            class="card__SubTitle-sc-1fzidng-3 cDBfHO"
          >
            Dallas
          </p>
          <p
            class="card__Text-sc-1fzidng-4 bcshOz"
          >
            J.R. Ewing, a Texas oil baron, uses manipulation and blackmail to achieve his ambitions, both business and personal. He often comes into conflict with his brother Bobby, his arch-enemy Cliff Barnes and his long-suffering wife Sue Ellen.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Card /> should render the <Card/> with populated data 1`] = `
<div
  class="card__Group-sc-1fzidng-2 cuQbQw"
>
  <div
    class="card__Container-sc-1fzidng-1 gcFCET"
  >
    <p
      class="card__Title-sc-1fzidng-0 ighwgO"
    >
      Comedy
    </p>
    <div
      class="card__Entities-sc-1fzidng-12 fjzXQn"
    >
      <div
        class="card__Item-sc-1fzidng-13 hkgFNq"
      >
        <img
          class="card__Image-sc-1fzidng-14 bjYtpL"
          src="/images/series/family/the-jeffersons/small.jpg"
        />
        <div
          class="card__Meta-sc-1fzidng-11 gJNAGL"
        >
          <p
            class="card__SubTitle-sc-1fzidng-3 cDBfHO"
          >
            The Jeffersons
          </p>
          <p
            class="card__Text-sc-1fzidng-4 bcshOz"
          >
            A nouveau riche, African-American family who move into a luxury apartment building develop close, if occasionally fractious, relationships with other tenants.
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="card__Container-sc-1fzidng-1 gcFCET"
  >
    <p
      class="card__Title-sc-1fzidng-0 ighwgO"
    >
      Drama
    </p>
    <div
      class="card__Entities-sc-1fzidng-12 fjzXQn"
    >
      <div
        class="card__Item-sc-1fzidng-13 hkgFNq"
      >
        <img
          class="card__Image-sc-1fzidng-14 bjYtpL"
          src="/images/series/drama/dallas/small.jpg"
        />
        <div
          class="card__Meta-sc-1fzidng-11 gJNAGL"
        >
          <p
            class="card__SubTitle-sc-1fzidng-3 cDBfHO"
          >
            Dallas
          </p>
          <p
            class="card__Text-sc-1fzidng-4 bcshOz"
          >
            J.R. Ewing, a Texas oil baron, uses manipulation and blackmail to achieve his ambitions, both business and personal. He often comes into conflict with his brother Bobby, his arch-enemy Cliff Barnes and his long-suffering wife Sue Ellen.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
